src/library/playlistfeature.cpp

```diff
@@ -113,7 +116,7 @@ bool PlaylistFeature::dropAcceptChild(const QModelIndex& index, QList<QUrl> urls
     // playlist.
     // pSource != nullptr it is a drop from inside Mixxx and indicates all
     // tracks already in the DB
-    QList<TrackId> trackIds = m_pTrackCollection->resolveTrackIdsFromUrls(urls,
+    QList<TrackId> trackIds = m_pLibrary->trackCollections()->internalCollection()->resolveTrackIdsFromUrls(urls,
```

Comment:
Adding redundant member variables should be avoided if possible, because they extend the *state* that needs to be managed (mentally). One option would be to introduce a utility function.

Only if repeatedly resolving multiple pointer indirections at runtime becomes too costly caching the result in an additional member variable would be justified.

I've tried to remove redundant member variables that I found to get a clearer view of the **essential dependencies**. I propose not revert this first small step, which would otherwise hinder future refactorings! Please keep in mind that the current state is considered work-in-progress.

Reply:
OK.